app/operations/story_operations.rb

```diff
@@ -0,0 +1,134 @@
+module StoryOperations
+
+  module MemberNotification
+    def self.included(base)
+      base.class_eval do
+        attr_reader :users_to_notify
+      end
+    end
+
+    def notify_users
+      return unless can_notify_mentioned_users?
+
+      notifier = Notifications.story_mention(model, users_to_notify)
+      notifier.deliver if notifier
+    end
+
+    def users_from_story
+      usernames = UsernameParser.parse(model.description)
+      return [] if usernames.empty?
+
+      @users_to_notify ||= model.users.where(username: usernames).all
+    end
+
+    def can_notify_mentioned_users?
+      model.description.present? && users_from_story.any? &&
+        !model.suppress_notifications
+    end
+  end
+
+  module StateChangeNotification
+    def notify_state_changed
+      return unless can_notify_state_changed?
+
+      notifier = Notifications.public_send(model.state.to_sym, model, model.acting_user)
+      notifier.deliver if notifier
+      IntegrationWorker.perform_async(model.project.id, integration_message)
+    end
+
+    def can_notify_state_changed?
+      return false unless model.state_changed?
+      return false if model.suppress_notifications
+      return false unless model.acting_user
+
+      actor = nil
+      case model.state
+      when 'started', 'delivered'
+        actor = model.requested_by
+        return false unless actor.email_delivery?
+      when 'accepted'
+        actor = model.owned_by
+        return false unless actor.email_acceptance?
+      when 'rejected'
+        actor = model.owned_by
+        return false unless actor.email_rejection?
+      else
+        return false
+      end
+
+      model.acting_user != actor
```

Comment:
na verdade isso tá certo, se for outro state tipo 'finished' é pra devolver false mesmo